### ISSUE TITLE
Add `deprecated-attribute` message

### ISSUE THREAD

**Issue by jacobtylerwalls** [pylint-dev/pylint#8855]
### Current problem

Similar to `deprecated-method`, we could have `deprecated-attribute` to warn for uses like this:

```py
>>> sqlite3.version_info
<stdin>:1: DeprecationWarning: version_info is deprecated and will be removed in Python 3.14
```

### Desired solution

`deprecated-attribute` message and then add attributes that were missed from python what's new documents, such as in 3.12:

- sqlite3.version
- sqlite3.version_info
- generator.throw
- agen.throw
- sys.last_type
- sys.last_value
- sys.last_traceback
- calendar.January
- calendar.February

### Additional context

The generator changes might be difficult to make, but the `sys` ones would be easy and high value because they were deprecated only in docs, not with code warnings.

**Comment by AmanSal1** [comment]
@jacobtylerwalls Can I work on this ?

**Comment by jacobtylerwalls** [comment]
Sure thing! Feel free to open a draft PR if you have questions about something along the way.

**Comment by AmanSal1** [comment]
@jacobtylerwalls Can you just sum up what all has to be done ?

**Comment by jacobtylerwalls** [comment]
Sure, you should take a look at the implementation of the other `deprecated-*` messages such as `deprecated-method`, look at how they are triggered, and figure out how to trigger a new similar message for attributes. You can then adapt the functional tests for `deprecated-method` in a very similar way. Basically, use your IDE to search for `deprecated-method` and just look at everything you find.

You'll probably need to trigger your check from `visit_attribute()` instead of `visit_call`.

**Comment by AmanSal1** [comment]
@jacobtylerwalls So till now what I have figured out is that:

1.DEPRECATED_ATTRIBUTE_MESSAGE = {
    "W4906": (
        "Using deprecated attribute %r",
        "deprecated-attribute",
        "The attribute is marked as deprecated and will be removed in the future.",
        {"old_names": [("WXXXX", "old-deprecated-attribute")], "shared": True},
    ),
}
Add this message in the DeprecatedMixin class and then make change the add a function called as deprecated_attributes () in it .

Can you tell is right  ?

**Comment by jacobtylerwalls** [comment]
Yep sounds right!

**Comment by AmanSal1** [comment]
@jacobtylerwalls  In the file "typecheck.py," the function "visit_attributes()" will also be modified because there are multiple functions with the same name in different files. I wanted to confirm with you if this understanding is correct.

**Comment by jacobtylerwalls** [comment]
That doesn't sound quite right, I think only the visit_attribute() method needs changing in the file with the deprecation checker. If it doesn't exist you'll need to create it (it's an override).

**Comment by AmanSal1** [comment]
"deprecated.py" In this file, there is no definition of 'visit_attributes()' method. I suppose I have to create it then?

**Comment by AmanSal1** [comment]
Or are you talking about the visit_attributes() method in the "class_checker.py" file ?

**Comment by jacobtylerwalls** [comment]
> @jacobtylerwalls  "deprecated.py" In this file, there is no definition of 'visit_attributes()' method. I suppose I have to create it then?

Yes, this is needed. No "s" at the end of attribute. If you check the parent classes you'll see it's an override.

**Comment by AmanSal1** [comment]
Wrote it by mistake . So I have to create a method with this name in deprecated.py as it is not there ?

**Comment by jacobtylerwalls** [comment]
Right, it will have the same signature as in the parent class.

**Comment by AmanSal1** [comment]
Okay got it !!

**Comment by AmanSal1** [comment]
@jacobtylerwalls Like I have made changes in the deprected.py I want to test them how can I test them and know if the changes are right or not. Like do I have to create a python file and then give to pylint something like that ?

**Comment by jacobtylerwalls** [comment]
You can take a look at https://pylint.readthedocs.io/en/stable/development_guide/contributor_guide/tests/index.html for those topics.

**Comment by AmanSal1** [comment]
@jacobtylerwalls For adding the test for the "attrribute" do we also need to update the "check_deprecate_method" or the other related methods because I am running a command "python -m pytest unittest_deprecated.py" and my 4 test are failing .Can you guide me little bit? 

![1](https://github.com/pylint-dev/pylint/assets/121633121/fe0345a7-a419-4727-ba63-da620f7f96dd)

**Comment by jacobtylerwalls** [comment]
At this point a draft PR would be easier to talk over, provides more info than a screen shot 👍

**Comment by AmanSal1** [comment]
Ok will draft soon

**Comment by AmanSal1** [comment]
Can you review it I accidently got some whitespace and all but can you review the logic ?
